Rewrite the 'TabPanel - Overview' description (#889)
DevExpress/devextreme-demos: JSDemos/Demos/TabPanel/Overview/description.md

> The [TabPanel](/Documentation/ApiReference/UI_Components/dxTabPanel/) UI component consists of the [Tabs](/Documentation/ApiReference/UI_Components/dxTabs/) and [MultiView](/Documentation/ApiReference/UI_Components/dxMultiView/) components. TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa. This demo shows the following TabPanel configuration::      

Reviewer: ```suggestion
The [TabPanel](/Documentation/ApiReference/UI_Components/dxTabPanel/) UI component consists of the [Tabs](/Documentation/ApiReference/UI_Components/dxTabs/) and [MultiView](/Documentation/ApiReference/UI_Components/dxMultiView/) components. TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa.
```

> The [TabPanel](/Documentation/ApiReference/UI_Components/dxTabPanel/) UI component consists of the [Tabs](/Documentation/ApiReference/UI_Components/dxTabs/) and [MultiView](/Documentation/ApiReference/UI_Components/dxMultiView/) components. TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa. This demo demonstrates the following TabPanel features:      

Reviewer 2: "TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa." - is it the same as "With TabPanel, users can press tabs to switch between views"?

Reviewer 2: "This demo demonstrates the following TabPanel **features**: ", but the headers below are not feature nouns. What if use such a structure:
```
To configure TabPanel, do the following: 
1) Bind the TabPanel to Data
2) Customize Tab Switch
```
?

Author: >"TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa." - is it the same as "With TabPanel, users can press tabs to switch between views"?

It is not completely the same. Yes, users can press tabs to switch between views but the can also swipe. Moreover, it is not the main point. TabPanel consists of dxTabs and dxMultiView, and one of the core TabPanel features is to synchronize a tab and a view's content.

>To configure TabPanel, do the following: 
1) Bind the TabPanel to Data
2) Customize Tab Switch

When we start a paragraph saying "To do X, do the following", we mean some purpose. To configure a component is not a purpose but something we should do to reach it. I would suggest something like 'This demo shows the following TabPanel configuration:'

>Customize Tab Switch

The word configure fits better since we do not customize (to make it convenient for somebody) but specify some properties.

### Bind the TabPanel to Data       

> The [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) property binds the TabPanel to data. To specify a custom template for all the TabPanel views, use the [itemTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTemplate). To do the same for the tabs, use [itemTitleTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTitleTemplate).

Reviewer 2: 1)Why use custom templates? The data objects use custom field names. This explanation would come in handy. The [Customize Item Appearance](https://js.devexpress.com/Documentation/Guide/UI_Components/TabPanel/Customize_Item_Appearance/) is an example.
2) What are the standard field names? As far as I could understand, they are listed in the items[] prop. description, but not the dataSource description (should dataSource be extended?). 
3) Should items[] be mentioned?

Author: Sorry, I did not understand your questions. 

1. Do you suggest to add something like 'You can define custom templates for data in TabPanel'? And what custom names use the data objects?
2. Which standard fields do you mean? Is it about the scheduler internal things that you described in the Resources demo? items and dataSource are alternatives to each other and developers cannot use them simultaneously.
3. No, see the previous point.

Reviewer 2: 1, 2) I meant to mention the default required item fields: `title` and `text` and also the ability to add additional fields (badge, icon, etc.). After that, write about itemTemplate и itemTitleTemplate in case the aforementioned options are not enough.
The dxTabPanel/Configuration/#dataSource article doesn't state that data objects should include fields from dxTabPanel/Configuration/items/. Should dxTabPanel/Configuration/#dataSource be extended?

Author: >I meant to mention the default required item fields: title and text

These items are not required, this is the point. Please see example https://codepen.io/alex_jameson/pen/WNRmOOb?editors=0010 

>The dxTabPanel/Configuration/#dataSource article doesn't state that data objects should include fields from dxTabPanel/Configuration/items/.

Because they don't have to.  Once again, these properties cannot be used simultaneously so they do not depend on each other.

Author: I guess that you confuse the items-dataSource pair in collection components(List, Toolbar, tabPanel) is similar to the columns-dataSource in grids. In the latter case, dataSource provides data and columns[] specifies which data to use. 

But the collection components work in a different way. You can specify data/layout only once, and to customize this data you need to use templates or specify markup directly using items[].

### Switch Between the Tabs     

> You can configure navigation between tabs. Specify the [swipeEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled) to define whether to switch between views with the swipe gesture. You can also [loop](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#loop) views and use the [animationEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#animationEnabled) property to specify whether to animate the change of the selected view.

Reviewer 2: ```suggestion
To switch between views with a swipe and animate the view change even on non-touch devices, set the [swipeEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled) and [animationEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#animationEnabled) properties to **true**, respectively. You can also enable [loop](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#loop) of views.

```

Reviewer 2: Add that `loop` occurs when a right/left arrow is used?

Author: The swipeEnabled and animationEnabled properties do not depend on each other. You can enable swipe but disable animation on any device and vice versa.

>You can also enable [loop](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#loop) of views.

We can say the same thing but shorter without the word 'enable'

Reviewer 2: All three properties are of equal priority. What if use a list?
```
To configure navigation between tabs, use the following properties:

* [swipeEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled)           
Allows users to swipe views.

...
```

Or a "property - default value - description" table?

Author: I think a list would be ok, but Roman suggested on the previous review iteration to transform the list to a paragraph. I guess that the reason behind that was inconsistency two parts of the description whose first part should is a paragraph.

Reviewer: No, I changed my mind, a list would be great. But it should after the selectedIndex paragraph so that the information about checkboxes go right after the list.

> To change the selected tab programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled) property that is the index of the currently selected tab in the [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) or [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) array. The [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#onSelectionChanged) allows you to specify a function that is executed when the selected tab changes. 

Reviewer: > To change the selected tab programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled) property that is the index of the currently selected tab in the [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) or [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) array.

I split the sentence in two, it was too long. Also, I removed links to the dataSource and items properties because they should be already linked in the section on data binding.

```suggestion
To switch between tabs programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#selectedIndex) property. It accepts the index of a tab in the **dataSource** or **items[]** array. The [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#onSelectionChanged) function allows you perform desired actions when the selected tab changes.
```

> To change the selected tab programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled) property that is the index of the currently selected tab in the [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) or [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) array. The [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#onSelectionChanged) allows you to specify a function that is executed when the selected tab changes. 

Reviewer 2: items[] is first mentioned here, but it would be good to mention it in the data binding part.

Author: I did not mention items in the data binding section because in this section we describe only properties that are present in this demo.

The [TabPanel](/Documentation/ApiReference/UI_Components/dxTabPanel/) UI component consists of the [Tabs](/Documentation/ApiReference/UI_Components/dxTabs/) and [MultiView](/Documentation/ApiReference/UI_Components/dxMultiView/) components. TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa.

### Bind the TabPanel to Data       

> The [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) property binds the TabPanel to data. You can use the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) API to shape, group, and otherwise modify data. You can also assign an array of TabPanel items to the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) property instead of **dataSource** as shown in this demo. Specify tab and view settings in this array.  

Reviewer: > The [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) property binds the TabPanel to data.

It is said as if there's no alternative.

> To specify a custom template for a tab or a view, use the [tabTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#tabTemplate) or [template](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#template) properties for an element in the **items[]** array.  To specify a custom template for all the TabPanel views, use the [itemTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTemplate). To do the same for the tabs, use [itemTitleTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTitleTemplate).

Reviewer: Info about default templates is missing.

> To specify a custom template for a tab or a view
> To specify a custom template for all the TabPanel views
> To do the same for the tabs

All three sentences have the same structure.
You must first designate the task you're going to solve. Don't start your paragraph with "To do the task, use X". Start with why a custom template is needed.

### Switch Between the Tabs     

To switch between tabs programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#selectedIndex) property. It accepts the index of a tab in the **dataSource** or **items[]** array. The [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#onSelectionChanged) function allows you perform desired actions when the selected tab changes.

> Use the following properties to configure navigation between tabs:       

Reviewer: ```suggestion
Use the following properties to configure user navigation between tabs:

```

- [swipeEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled)      

> Defines whether to switch between views with the swipe gesture.

Reviewer: ```suggestion
Defines whether to switch between views with the swipe gesture.

```

- [loop](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#loop)      

> Specifies whether to loop views.

Reviewer: ```suggestion
Specifies whether to loop views.

```

> The [TabPanel](/Documentation/ApiReference/UI_Components/dxTabPanel/) UI component consists of the [Tabs](/Documentation/ApiReference/UI_Components/dxTabs/) and [MultiView](/Documentation/ApiReference/UI_Components/dxMultiView/) components. TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa.

Reviewer 3: ```suggestion
The [TabPanel](/Documentation/ApiReference/UI_Components/dxTabPanel/) UI component consists of [Tabs](/Documentation/ApiReference/UI_Components/dxTabs/) and [MultiView](/Documentation/ApiReference/UI_Components/dxMultiView/) components. The TabPanel automatically synchronizes the selected tab with the currently displayed view and vice versa.
```

> ### Bind the TabPanel to Data       

Reviewer: ```suggestion
### Generate Similar Tabs Based on a Data Source       
```

> ### Generate Similar Tabs Based on a Data Source       

Reviewer 4: Similar to what? Do you mean similar to each other? Is it possible to say that they are based on a template?

Author: 1. Yes, they all have the same layout.
2. Not really. We have multiple properties that contain the word 'template' in both sections of this text and this circumstance can be confusing.

Reviewer 4: I see. Then we can leave this as is, for it seems I can't come up with something better.

> To generate similar tabs and views, bind the TabPanel to data. Use the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) or [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) property to do this. Both these properties can work with local arrays, but **dataSource** also accepts a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) object. You can use this object if you need to process the local array or fetch the array from a remote data source. In this demo, the **dataSource** property is set to a local array.

Reviewer 3: ```suggestion
To generate similar tabs and views, bind the TabPanel to data. Use the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) [which items?] or [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) property to do this. Both these properties can work with local arrays, but the **dataSource** also accepts a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) object. You can use this object if you need to process the local array or fetch the array from a remote data source. In this demo, the **dataSource** property is set to a local array.
```

Author: This is a property's name with a link to it - [items[]](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/)

Reviewer 3: ok

> You can assign an array of TabPanel items to the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) property as shown in this demo. Each item in this array configures an individual tab and an associated view. When these items are objects, use fields of these objects to specify tab and view properties listed in the **items[]** section. Each item of the *"String"* type in this array creates a tab and a view with this string.
> 
> You can assign the same array to the [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) propertyinstead of **items[]**. The [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) API allows you to shape, group, and otherwise modify data. 

Reviewer: ```suggestion
To generate similar tabs and views, bind the TabPanel to data. Use the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) or [dataSource](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#dataSource) property to do this. Both these properties can work with local arrays, but **dataSource** also accepts a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) object. You can use this object if you need to process the local array or fetch the array from a remote data source. In this demo, the **dataSource** property is set to a local array.

Each object in the **items[]** or **dataSource** array can contain predefined fields, such as **title** or **icon** (see the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) section for the full list). The TabPanel automatically recognizes these fields and creates default tab and view appearance based on them. However, if your data objects contain custom fields, you need to specify the [itemTitleTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTitleTemplate) and [itemTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTemplate) that define how to display the fields in tabs and views. This demo illustrates the latter use case.
```

> Each object in the **items[]** or **dataSource** array can contain predefined fields, such as **title** or **icon** (see the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) section for the full list). The TabPanel automatically recognizes these fields and creates default tab and view appearance based on them. However, if your data objects contain custom fields, you need to specify the [itemTitleTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTitleTemplate) and [itemTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTemplate) that define how to display the fields in tabs and views. This demo illustrates the latter use case.

Reviewer 4: ```suggestion
Each object in the **items[]** or **dataSource** array can contain predefined fields, such as **title** or **icon** (see the [items[]](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/) section for the full list). The TabPanel automatically recognizes these fields and creates the default tab and view appearance based on them. However, if your data objects contain custom fields, you need to specify the [itemTitleTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTitleTemplate) and [itemTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTemplate) that define how to display the fields in tabs and views. This demo illustrates the latter use case.
```

> ### Create Individual Tabs

Reviewer 4: I believe these are contraposed to "similar" tabs mentioned earlier. While the "similar" tabs are template-based, these have custom layout, am I right?

Author: Please see my comment about 'template-based' above.

Reviewer 4: Yeah, I see. No more questions then.

> If you need to specify a single template for all the TabPanel views, use the [itemTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTemplate). To do the same for the tabs, use [itemTitleTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#itemTitleTemplate).

Reviewer: ```suggestion
### Create Individual Tabs

If you want each tab and view to have differently sctructured content, define individual templates. To do this, assign an array of objects to the **items[]**  or **dataSource** property and specify the [tabTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#tabTemplate) and [template](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#template) properties in each object. This use case is illustrated in the following tutorial: [Getting Started with TabPanel]().
```

> If you want each tab and view to have differently sctructured content, define individual templates. To do this, assign an array of objects to the **items[]**  or **dataSource** property and specify the [tabTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#tabTemplate) and [template](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#template) properties in each object. This use case is illustrated in the following tutorial: [Getting Started with TabPanel]().

Reviewer 4: ```suggestion
If you want each tab and view to have differently structured content, define individual templates. To do this, assign an array of objects to the **items[]** or **dataSource** property and specify the [tabTemplate](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#tabTemplate) and [template](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/items/#template) properties in each object. This use case is illustrated in the following tutorial: [Getting Started with TabPanel]().
```
P.S. An array of which objects should I assign to the mentioned properties?

Author: There are lots of options. These objects can have predefined structure and the TabPanel will create valid items automatically, The objects can also be have any fields but then users have to specify itemTemplae / itemTitleTemplate (see line 7)

> ### Switch Between the Tabs     

Reviewer 4: Do you really need this emphasis? This is just a way to switch between tabs. The way is important while tabs are not.

```suggestion
### Switch Between Tabs
```

> To switch between tabs programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#selectedIndex) property. It accepts the index of a tab in the **dataSource** or **items[]** array. The [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#onSelectionChanged) function allows you perform desired actions when the selected tab changes.

Reviewer 3: ```suggestion
To switch between tabs programmatically, use the [selectedIndex](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#selectedIndex) property. It accepts the index of a tab in the **dataSource** or **items[]** array. The [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#onSelectionChanged) function allows you to perform the desired actions when the selected tab changes.
```

Use the following properties to configure user navigation between tabs:

- [swipeEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#swipeEnabled)      

> Defines whether to switch between views with a swipe gesture.

Reviewer 3: ```suggestion
Defines whether or not to switch between views with a swipe gesture.
```

Author: Hello. We have already discussed this phrase a few months ago (please see the screenshots or the [discussion in Teams](https://teams.microsoft.com/l/message/19:b8c5aa2758c44e2db10312fae66683f2@thread.skype/1617040695260?tenantId=e4d60396-9352-4ae8-b84c-e69244584fa4&groupId=9298adba-77e3-4b3e-a0eb-b3c426b4ba6d&parentMessageId=1617005840357&teamName=DX%20Tech%20Writers%20Club&channelName=doc.guidelines&createdTime=1617040695260)). Has anything changed? 
![image](https://user-images.githubusercontent.com/33040934/122260334-d284a480-cedb-11eb-9495-9f423f9c2da4.png)
![image](https://user-images.githubusercontent.com/33040934/122260348-d6b0c200-cedb-11eb-966f-1f3578f3bf1e.png)

Reviewer 3: Yes I know. But with the word " between " there, I think the sentence works better this way. but i don't mind if you don't keep the addition.

> Defines whether to switch between views with the swipe gesture.

Reviewer 4: Do you support only a single specific swipe gesture that does not allow any freedom? I suppose you meant **any** acceptable swipe gesture (swipe right or left, for instance). I could be wrong, however.

```suggestion
Defines whether to switch between views with a swipe gesture.
```

Author: Good point

- [loop](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#loop)      
Specifies whether to loop views.

- [animationEnabled](/Documentation/ApiReference/UI_Components/dxTabPanel/Configuration/#animationEnabled)      

> Specifies whether to animate the change of the selected view.

Reviewer 4: Did you mean "to animate transition between views" (that is, create a visual switch/transition process)?

Author: Yes, exactly.
